pkg/controllers/subnet/namespace_handler.go

```diff
@@ -0,0 +1,94 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package subnet
+
+import (
+	"context"
+	"reflect"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/util"
+)
+
+// Subnet controller should watch event of namespace, when there are some updates of namespace labels,
+// controller should build tags and update VpcSubnet according to new labels.
+
+type EnqueueRequestForNamespace struct {
+	Client client.Client
+}
+
+func (e *EnqueueRequestForNamespace) Create(_ event.CreateEvent, _ workqueue.RateLimitingInterface) {
+	log.V(1).Info("namespace create event, do nothing")
+}
+
+func (e *EnqueueRequestForNamespace) Delete(_ event.DeleteEvent, _ workqueue.RateLimitingInterface) {
+	log.V(1).Info("namespace delete event, do nothing")
+}
+
+func (e *EnqueueRequestForNamespace) Generic(_ event.GenericEvent, _ workqueue.RateLimitingInterface) {
+	log.V(1).Info("namespace generic event, do nothing")
+}
+
+func (e *EnqueueRequestForNamespace) Update(updateEvent event.UpdateEvent, l workqueue.RateLimitingInterface) {
+	obj := updateEvent.ObjectNew.(*v1.Namespace)
+	if isInSysNs, err := util.IsSystemNamespace(nil, "", obj); err != nil {
```

Comment:
I guess we cannot use `IsSystemNamespace` any more, it is for T1 networking. We need to check if it has infra annotation? @lxiaopei please help confirm

Reply:
In VPC mode, the system namespaces are annotated with "nsx.vmware.com/vpc_network_config: <supervisor ID>".

Reply:
is system namespace has different collection between t1 and vpc mode? i mean if in vpc mode, system namespace do not have the system annotation on namespace anymore?

Reply:
yes. In VPC mode, the system namespaces are annotated with "nsx.vmware.com/vpc_network_config: "

Reply:
As discussed, defer it.